### PR TITLE
Improvement/eip adapter

### PR DESCRIPTION
**Pull request by schaebo**
**Changes**
Improve Ethernet/IP adapter to work for Allen-Bradley ControlLogix and CompactLogix devices

### CODE REVIEW

File: hivemq-edge/src/main/java/com/hivemq/api/resources/impl/ProtocolAdaptersResourceImpl.java [hunk 1]
```diff
@@ -81,20 +71,28 @@ public ProtocolAdaptersResourceImpl(
...
+        Set<ProtocolAdapter> installedAdapters =
+                protocolAdapterManager.getAllAvailableAdapterTypes().values().stream().map(installedAdapter -> {
+                    try {
+                        return ProtocolAdapterApiUtils.convertInstalledAdapterType(objectMapper,
+                                protocolAdapterManager,
+                                installedAdapter,
+                                configurationService);
+                    } catch (Throwable t) {
+                        logger.warn("Not able to properly load protocol adapter", t);
+                        return null;
```
Comment (DC2-DanielKrueger) [inline hivemq-edge/src/main/java/com/hivemq/api/resources/impl/ProtocolAdaptersResourceImpl.java:83]: What happens if you return a null in a map() function?

File: modules/hivemq-edge-module-plc4x/src/main/java/com/hivemq/edge/adapters/plc4x/types/eip/EIPAdapterConfig.java [hunk 2]
```diff
@@ -19,81 +19,86 @@
...
-import java.time.LocalDate;
-import java.time.LocalDateTime;
-import java.time.LocalTime;
 import java.util.ArrayList;
 import java.util.List;
 
 
 public class EIPAdapterConfig extends Plc4xAdapterConfig {
 
+    @JsonProperty("port")
```
Comment (DC2-DanielKrueger) [inline modules/hivemq-edge-module-plc4x/src/main/java/com/hivemq/edge/adapters/plc4x/types/eip/EIPAdapterConfig.java:31]: Just curious: Wasnt a port needed before?

Comment (schaebo) [reply modules/hivemq-edge-module-plc4x/src/main/java/com/hivemq/edge/adapters/plc4x/types/eip/EIPAdapterConfig.java:31]: The parent class already contains a port, just overridden to add a default value here.

File: hivemq-edge/src/main/java/com/hivemq/api/resources/impl/ProtocolAdaptersResourceImpl.java (hunk 1 above)
Comment (schaebo) [reply hivemq-edge/src/main/java/com/hivemq/api/resources/impl/ProtocolAdaptersResourceImpl.java:83]: the stream then contains a null value at this place, that get's then removed by the `filter(Objects::nonNull)` afterwards

**Review by simon622 — APPROVED** [review]
Looks good

**Review by DC2-DanielKrueger — APPROVED** [review]

**Comment by sfrehse** [comment]
I will merge it, the build itself is actually green.

**Pull request merged**